front-end/src/components/Profile/ProfileTabs/ProfileTabs.jsx

```diff
@@ -28,6 +26,20 @@ const tabs = [
 const ProfileTabs = () => {
   const { state } = useLocation();
   const [selectedTab, setSelectedTab] = useState(state);
+  const [tabs, setTabs] = useState(defaultTabs);
+  const { currentUser } = useContext(CurrentUserContext);
+
+  useEffect(() => {
+    if (currentUser.role === 'teacher') {
+      setTabs([
```

Comment:
If you change state and depend on previous state, you better use a callback (prev) => ({...prev, newValue}) inside of setState function.